pynamodb/_compat.py

```diff
@@ -1,10 +1,6 @@
-import six
 import sys
 
-if six.PY2:
-    from inspect import getargspec as getfullargspec
-else:
-    from inspect import getfullargspec
+from inspect import getfullargspec
 
 
 class FakeGenericMeta(type):
```

Comment:
From Python 3.7 we can use real `typing.Generic`. Classes derived from `Generic` don't require the type params, so existing code will be unaffected.
```
>>> from typing import *
>>> T = TypeVar('T')
>>> class MyBase(Generic[T]): pass
...
>>> class ThisIsOk(MyBase[int]): pass
...
>>> class ThisIsAlsoOk(MyBase): pass
...
```
However, IIRC Python 3.6's `Generic` doesn't like coexisting with other meta classes :( In fact, if following this modernization we don't have .pyi files anymore, we might need to branch out with an `if TYPE_CHECKING` while defining class Index.